hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/MorphingPortableReaderTest.java

```diff
@@ -0,0 +1,218 @@
+        int aByte = reader.readInt("byte");
+        int aShort = reader.readInt("short");
+        int aChar = reader.readInt("char");
+        int aInt = reader.readInt("int");
+
+        assertEquals(1, aByte);
+        assertEquals(3, aShort);
+        assertEquals(2, aChar);
+        assertEquals(4, aInt);
+        assertEquals(0, reader.readInt("NO SUCH FIELD"));
+
+    }
+
+    @Test
+    public void testReadLong() throws Exception {
+        long aByte = reader.readLong("byte");
+        long aShort = reader.readLong("short");
+        long aChar = reader.readLong("char");
+        long aInt = reader.readLong("int");
+
+        assertEquals(1, aByte);
+        assertEquals(3, aShort);
+        assertEquals(2, aChar);
+        assertEquals(4, aInt);
+        assertEquals(0, reader.readLong("NO SUCH FIELD"));
+
+    }
+
+    @Test
+    public void testReadUTF() throws Exception {
+        assertNull(reader.readUTF("NO SUCH FIELD"));
+    }
+
+    @Test
+    public void testReadBoolean() throws Exception {
+        assertFalse(reader.readBoolean("NO SUCH FIELD"));
+    }
+
+    @Test
+    public void testReadByte() throws Exception {
+        assertEquals(0, reader.readByte("NO SUCH FIELD"));
+    }
+
+    @Test
+    public void testReadChar() throws Exception {
+        assertEquals(0, reader.readChar("NO SUCH FIELD"));
+    }
+
+    @Test
+    public void testReadDouble() throws Exception {
+        double aByte = reader.readDouble("byte");
+        double aShort = reader.readDouble("short");
+        double aChar = reader.readDouble("char");
+        double aInt = reader.readDouble("int");
+        double aFloat = reader.readDouble("float");
+        double aDouble = reader.readDouble("double");
+
+        assertEquals(1, aByte, 0);
+        assertEquals(3, aShort, 0);
+        assertEquals(2, aChar, 0);
+        assertEquals(4, aInt, 0);
+        assertEquals(1f, aFloat, 0);
+        assertEquals(2d, aDouble, 0);
+        assertEquals(0, reader.readDouble("NO SUCH FIELD"), 0);
+    }
+
+    @Test
+    public void testReadFloat() throws Exception {
+        float aByte = reader.readFloat("byte");
+        float aShort = reader.readFloat("short");
+        float aChar = reader.readFloat("char");
+        float aInt = reader.readFloat("int");
+        float aFloat = reader.readFloat("float");
+
+        assertEquals(1, aByte, 0);
+        assertEquals(3, aShort, 0);
+        assertEquals(2, aChar, 0);
+        assertEquals(4, aInt, 0);
+        assertEquals(1f, aFloat, 0);
+        assertEquals(0, reader.readFloat("NO SUCH FIELD"), 0);
+
+    }
+
+    @Test
+    public void testReadShort() throws Exception {
+        int aByte = reader.readShort("byte");
+        int aShort = reader.readShort("short");
+
+        assertEquals(1, aByte);
+        assertEquals(3, aShort);
+        assertEquals(0, reader.readShort("NO SUCH FIELD"));
+
+    }
+
+    @Test(expected = IncompatibleClassChangeError.class)
```

Comment:
I'm confused with the below tests. They all expect an exception to be thrown, which is perfectly fine. But they also all have an assert as the last line. Afaik this assert will never be executed.